Xamarin.Forms.Platform.Android/CollectionView/ItemsViewRenderer.cs

```diff
@@ -515,13 +515,11 @@ protected virtual void ReconcileFlowDirectionAndLayout()
 
 			var effectiveFlowDirection = ((IVisualElementController)Element).EffectiveFlowDirection;
 
-			if (effectiveFlowDirection.IsRightToLeft() && !linearLayoutManager.ReverseLayout)
+			if (effectiveFlowDirection.IsRightToLeft() && !linearLayoutManager.ReverseLayout && Context.GetActivity().Window.DecorView.LayoutDirection == LayoutDirection.Ltr)
 			{
 				linearLayoutManager.ReverseLayout = true;
-				return;
-			}
-
-			if (effectiveFlowDirection.IsLeftToRight() && linearLayoutManager.ReverseLayout)
+			} 
+			else if (effectiveFlowDirection.IsLeftToRight() && linearLayoutManager.ReverseLayout && Context.GetActivity().Window.DecorView.LayoutDirection == LayoutDirection.Rtl)
```

Comment:
This probably won't work if `Window.DecorView.LayoutDirection` is LTR and the user changes flow direction to LTR from RTL.

Reply:
> This probably won't work if `Window.DecorView.LayoutDirection` is LTR and the user changes flow direction to LTR from RTL.

There are 4 possibilities, and all are taken care of:

LayoutDirection | FlowDirection | Need to Reverse
-- | -- | --
Ltr | RightToLeft | Y
Ltr | LeftToRight | N
Rtl | RightToLeft | N
Rtl | LeftToRight | Y

When **Need to Reverse** is **N** then the system will take care of it.

Reply:
My initial reaction was if we set `Window.DecorView.LayoutDirection` to LTR in `MainActivity` (which should be default in most cases) and effective flow to RTL, then the layout is reversed successfully. 

If we then dynamically change effective flow to LTR, then there is no if condition to satisfy this request because `Window.DecorView.LayoutDirection` is still LTR. Do you mean the system will take care of this automatically?

@samhouts We should add a manual test for this, demonstrating all possibilities though it'll be a bit tricky with manifest and activity changes.